master/internal/api_workspace.go

```diff
@@ -66,36 +67,46 @@ func maskStorageConfigSecrets(w *workspacev1.Workspace) error {
 	return nil
 }
 
-func (a *apiServer) validateRequestClusterName(clusterName string) error {
+func (a *apiServer) validateRequestClusterName(clusterName string) (string, error) {
 	if len(clusterName) == 0 && len(a.m.allRms) > 1 {
-		return status.Errorf(codes.InvalidArgument, "must specify a cluster name when using MultiRM")
+		return "", status.Errorf(codes.InvalidArgument, "must specify a cluster name when using MultiRM")
+	}
+
+	if len(a.m.allRms) == 1 && clusterName == "" {
+		for k := range a.m.allRms {
+			clusterName = k
+		}
 	}
 	_, ok := a.m.allRms[clusterName]
 	if !ok {
-		return status.Errorf(codes.InvalidArgument,
+		return "", status.Errorf(codes.InvalidArgument,
 			"no resource manager with cluster name %s", clusterName)
 	}
-	return nil
+	return clusterName, nil
 }
 
-func (a *apiServer) validateClusterNamespacePairs(clusterNamespacePairs map[string]string) error {
+func (a *apiServer) validateClusterNamespacePairs(clusterNamespacePairs map[string]string) (map[string]string, error) {
 	allClusters := make(map[string]int)
 	for clusterName, namespace := range clusterNamespacePairs {
 		if _, ok := allClusters[clusterName]; ok {
-			return status.Errorf(codes.InvalidArgument, `Cannot specify the same cluster name with 
+			return nil, status.Errorf(codes.InvalidArgument, `Cannot specify the same cluster name with 
 			different namespace, workspace-namespace bindings are unique per cluster per workspace.`)
 		}
 		allClusters[clusterName] = 1
 		if len(clusterName) > 0 && len(namespace) == 0 {
-			return status.Errorf(codes.InvalidArgument,
+			return nil, status.Errorf(codes.InvalidArgument,
 				"Must specify a Kubernetes namespace")
 		}
-		err := a.validateRequestClusterName(clusterName)
+		newClusterName, err := a.validateRequestClusterName(clusterName)
 		if err != nil {
-			return err
+			return nil, err
+		}
+		if newClusterName != clusterName {
+			clusterNamespacePairs[newClusterName] = namespace
+			delete(clusterNamespacePairs, clusterName)
```

Comment:
Can we add a comment here that's along the lines of 
`// This might occur when using singleRM with a cluster name defined in the master config, but not specified in a given request`
Kind of similar to the comment above, but good to clarify for anyone trying to read the code

Reply:
done